### PR TITLE
Rewrite `connectModule` and `ModuleProvider`.

### LINKED ISSUE #90: Rewrite `connect-module`.
It is mostly taken from `react-redux`, but it was when they were in the middle of working out that selector-based rewrite.  This is mostly done, but #86 blocks testing.

### PR DESCRIPTION
**Pull request by wtgtybhertgeghgtwtg**
Closes #90.  It's not breaking any tests and the examples aren't throwing more errors than they usually do, but it might be a good idea to up the coverage before releasing this.

### CODE REVIEW

File: src/connectModule/connectModules.js [hunk 1]
```diff
@@ -164,7 +97,7 @@ export default function connectModules(mapStateToProps, modules) {
         this.subscription = new Subscription(
           this.store,
           this.parentSub,
-          onStoreStateChange.bind(this));
+          this::onStoreStateChange);
```
Comment (mboperator) [inline src/connectModule/connectModules.js]: This is a pretty cool operator. Is this part of babel stage-0?

File: src/connectModule/createFinalPropsSelector.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import deepAssign from 'deep-assign';
+
+import createGetState from './createMapStateToProps';
+
+export default function createFinalPropsSelector(mapStateToProps, mapDispatchToProps) {
```
Comment (mboperator) [inline src/connectModule/createFinalPropsSelector.js]: Are these more or less the same as in the `react-redux` library?

Comment (wtgtybhertgeghgtwtg) [inline src/connectModule/createFinalPropsSelector.js]: From the `5.0.0` beta, yeah.  If you'll recall, `ModuleProvider` used to just wrap `react-redux`, but that ended up having a few issues for one reason or another.  Once `react-redux@5.0.0` is launched, I'd like to try and move back to that.

Comment (wtgtybhertgeghgtwtg) [inline src/connectModule/createFinalPropsSelector.js]: Sure.

Comment (wtgtybhertgeghgtwtg) [inline src/connectModule/createFinalPropsSelector.js]: I'm going to hold off on this until `react-redux` has some benchmarks with which to compare.

**Pull request merged**